async-http-client/src/main/scala/org/http4s/client/asynchttpclient/AsyncHttpClient.scala

```diff
@@ -40,15 +54,7 @@ object AsyncHttpClient {
     */
   def resource[F[_]](config: AsyncHttpClientConfig = defaultConfig)(
       implicit F: ConcurrentEffect[F]): Resource[F, Client[F]] =
-    Resource
-      .make(F.delay(new DefaultAsyncHttpClient(config)))(c => F.delay(c.close()))
-      .map(client =>
-        Client[F] { req =>
-          Resource(F.async[(Response[F], F[Unit])] { cb =>
-            client.executeRequest(toAsyncRequest(req), asyncHandler(cb))
-            ()
-          })
-      })
+    Resource.make(allocate(config))(_._2).map(_._1)
```

Comment:
Isn't this just `Resource.apply(allocate(config))`?

Reply:
Yup, learned more about resource construction, thanks!